### PR TITLE
fix: Add latency-check for DERP over HTTP(s)

### PR DESCRIPTION
**Pull request by kylecarbs**
This fixes scenarios where latency wasn't being reported if
a connection had UDP entirely blocked.

### CODE REVIEW

**Review by ammario — APPROVED** [review]

**Comment by kylecarbs** [comment]
This also enables keep alives. The connection was timing out every ~2mins.

**Review by coadler — APPROVED** [review]

File: tailnet/coordinator.go [hunk 1]
```diff
@@ -40,7 +40,10 @@ func ServeCoordinator(conn net.Conn, updateNodes func(node []*Node) error) (func
 			}
 			err = updateNodes(nodes)
 			if err != nil {
-				errChan <- xerrors.Errorf("update nodes: %w", err)
+				select {
```
Comment (mafredri) [inline tailnet/coordinator.go:43]: Perhaps we should make the `errChan` only of size 1 (to make the behavior/expectation explicit) and maybe create a function called `sendErr()` that does the `select`. Right now I suppose these errors could fill up the `errChan` and then the `Decode` above would still hang?

**Review by mafredri — APPROVED** [review]
Looks alright although unfamiliar with most of it since I didn't review the original Tailscale PR 😅. Small change to `ServeCoordinator` could still be warranted though.

File: tailnet/coordinator.go (hunk 1 above)
Comment (kylecarbs) [reply tailnet/coordinator.go:43]: Good idea!

**Review by mafredri — APPROVED** [review]

**Pull request merged**